commit d28dc711217a2d5cafb64ec4e33a469b01376d91
Author: Gaston, Jason D <jason.d.gaston@intel.com>
Date:   Sun Feb 24 20:03:42 2008 +0100

    i2c-i801: Add support for the ICH10
    
    Add the Intel ICH10 SMBus Controller DeviceID's and updates
    Tolapai support.
    
    Signed-off-by: Jason Gaston <jason.d.gaston@intel.com>
    Signed-off-by: Jean Delvare <khali@linux-fr.org>

Backported to Debian's 2.6.24 by dann frazier <dannf@hp.com>

diff -urpN linux-source-2.6.24.orig/Documentation/i2c/busses/i2c-i801 linux-source-2.6.24/Documentation/i2c/busses/i2c-i801
--- linux-source-2.6.24.orig/Documentation/i2c/busses/i2c-i801	2008-01-24 15:58:37.000000000 -0700
+++ linux-source-2.6.24/Documentation/i2c/busses/i2c-i801	2008-05-01 17:27:17.000000000 -0600
@@ -12,8 +12,9 @@ Supported adapters:
   * Intel 82801G (ICH7)
   * Intel 631xESB/632xESB (ESB2)
   * Intel 82801H (ICH8)
-  * Intel ICH9
+  * Intel 82801I (ICH9)
   * Intel Tolapai
+  * Intel ICH10
    Datasheets: Publicly available at the Intel website
 
 Authors: 
diff -urpN linux-source-2.6.24.orig/drivers/i2c/busses/Kconfig linux-source-2.6.24/drivers/i2c/busses/Kconfig
--- linux-source-2.6.24.orig/drivers/i2c/busses/Kconfig	2008-01-24 15:58:37.000000000 -0700
+++ linux-source-2.6.24/drivers/i2c/busses/Kconfig	2008-05-01 17:27:17.000000000 -0600
@@ -177,6 +177,8 @@ config I2C_I801
 	    ESB2
 	    ICH8
 	    ICH9
+	    Tolapai
+	    ICH10
 
 	  This driver can also be built as a module.  If so, the module
 	  will be called i2c-i801.
diff -urpN linux-source-2.6.24.orig/drivers/i2c/busses/i2c-i801.c linux-source-2.6.24/drivers/i2c/busses/i2c-i801.c
--- linux-source-2.6.24.orig/drivers/i2c/busses/i2c-i801.c	2008-01-24 15:58:37.000000000 -0700
+++ linux-source-2.6.24/drivers/i2c/busses/i2c-i801.c	2008-05-01 17:32:48.000000000 -0600
@@ -35,6 +35,8 @@
     ICH8		283E
     ICH9		2930
     Tolapai		5032
+    ICH10		3A30
+    ICH10		3A60
     This driver supports several versions of Intel's I/O Controller Hubs (ICH).
     For SMBus support, they are similar to the PIIX4 and are part
     of Intel's '810' and other chipsets.
@@ -545,6 +547,8 @@ static struct pci_device_id i801_ids[] =
 	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ICH8_5) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ICH9_6) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_TOLAPAI_1) },
+	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ICH10_4) },
+	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ICH10_5) },
 	{ 0, }
 };
 
@@ -566,6 +570,8 @@ static int __devinit i801_probe(struct p
 	case PCI_DEVICE_ID_INTEL_ICH8_5:
 	case PCI_DEVICE_ID_INTEL_ICH9_6:
 	case PCI_DEVICE_ID_INTEL_TOLAPAI_1:
+	case PCI_DEVICE_ID_INTEL_ICH10_4:
+	case PCI_DEVICE_ID_INTEL_ICH10_5:
 		isich4 = 1;
 		break;
 	default:
